koa-crud/test/integration/data-access/vendor.spec.ts

```diff
@@ -0,0 +1,360 @@
+      it('should successfully add vendor to the database', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await expect(insertOneVendor(mock)).to.eventually.fulfilled;
+      });
+    });
+
+    describe('GIVEN no name', () => {
+      it('should throw an error', async () => {
+        const mock = {
+          name: '',
+          type: VendorType.Seamless,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await expect(insertOneVendor(mock)).to.eventually.rejected;
+      });
+    });
+
+    describe('GIVEN no type', () => {
+      it('should throw an error', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: '',
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await expect(insertOneVendor(mock)).to.eventually.rejected;
+      });
+    });
+
+    describe('GIVEN invalid type', () => {
+      it('should throw an error', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: 'qwe',
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await expect(insertOneVendor(mock)).to.eventually.rejected;
+      });
+    });
+  });
+
+  describe('Vendor exists? By filters', () => {
+    afterEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    beforeEach(() => {
+      return Vendor.deleteMany({});
+    });
+
+    describe('GIVEN existent filters', () => {
+      it('should return true', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        const main = await insertOneVendor(mock);
+        await expect(vendorExistsByFilter({ _id: main._id })).to.eventually
+          .fulfilled.be.true;
+      });
+    });
+
+    describe('GIVEN non existent filters', () => {
+      it('should return false', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await insertOneVendor(mock);
+        await expect(vendorExistsByFilter({ _id: mockedId })).to.eventually
+          .fulfilled.be.false;
+      });
+    });
+
+    describe('GIVEN invalid filters', () => {
+      it('should throw an error', async () => {
+        const mock = {
+          name: 'Luis Angelo Belmonte',
+          type: VendorType.Transfer,
+          dateTimeCreated: new Date(),
+          dateTimeUpdated: new Date(),
+        };
+        await insertOneVendor(mock);
+        await expect(vendorExistsByFilter({ _id: 'qwe' })).to.eventually
+          .rejected;
```

Comment:
If you want to assert the message thrown then you can use `rejectedWith('')`

Reply:
Noted on this sir.